backend/src/utils/sanitize.ts

```diff
@@ -0,0 +1,31 @@
+export const sanitizeCommitMessage = (inputStr: string):string => {
+  let str = inputStr.trim()
+  str = stripDangerousSpecialChars(str)
+  str = stripConsecutiveSpecialChars(str)
+  return str
+}
+  
+export const sanitizeBranchName = (inputStr: string):string => {
+  // Rules for branch names: https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html
+  let str = inputStr.trim()
+  str = stripAllSpecialCharsButDash(str)
+  str = stripConsecutiveSpecialChars(str)
+  str = stripStartingDashFromString(str)
+  return str
+}
```

Comment:
Would use something cleaner here at some point. Maybe some `reduce` magic could be used in the future.

Reply:
I don't like using the let. I'd recommend we go more into functional programming and use `pipe` [(More info)](https://www.freecodecamp.org/news/pipe-and-compose-in-javascript-5b04004ac937/). We could write our own generic pipe function or use  functional javascript library like ramda or lodash